eng/packages/http-client-csharp/generator/Azure.Generator/src/NamespaceVisitor.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.TypeSpec.Generator.ClientModel;
+using Microsoft.TypeSpec.Generator.ClientModel.Providers;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Providers;
+
+namespace Azure.Generator
+{
+    internal class NamespaceVisitor : ScmLibraryVisitor
+    {
+        protected override ModelProvider? Visit(InputModelType model, ModelProvider? type)
+        {
+            if (type is not null)
+            {
+                UpdateNamespace(type);
+            }
+            return type;
+        }
+
+        protected override TypeProvider? Visit(InputEnumType enumType, TypeProvider? type)
+        {
+            if (enumType.Usage.HasFlag(InputModelTypeUsage.ApiVersionEnum))
+            {
+                return type;
+            }
+
+            if (type is not null)
+            {
+                UpdateNamespace(type);
+            }
+            return type;
+        }
+
+        protected override TypeProvider? Visit(TypeProvider type)
+        {
+            if (type is EnumProvider && type.Name == "ServiceVersion")
+            {
+                return type;
+            }
+
+            if (type is ModelProvider || type is EnumProvider || type is ModelFactoryProvider
+                || type is MrwSerializationTypeDefinition || type is FixedEnumSerializationProvider || type is ExtensibleEnumSerializationProvider)
+            {
+                UpdateNamespace(type);
+            }
+            return type;
+        }
+
+        private static void UpdateNamespace(TypeProvider type)
+        {
+            // TODO: need to take consideration of model-namespace configuration
+            // if model-namespace is false, keep the default namespace
+            // if model-namespace is true, append ".Models" to the namespace
+            if (!type.Type.Namespace.EndsWith("Models"))
+            {
+                type.Type.Namespace = $"{type.Type.Namespace}.Models";
+            }
```

Comment:
I think we are not changing them in a correct path.
the namespaces in our azure libraries are like this:
first we have a RootNamespace which comes from the configuration.
If the class is a model or enum or model factory, it should be `RootNamespace.Models`
all other class should just be `RootNamespace`.
Now in this visitor, quite a few models are not changed, this is not entirely correct

Reply:
This is the method only affecting model or enum or model factory, please check the caller of this method

Reply:
updated